python/plugins/processing/algs/qgis/SinglePartsToMultiparts.py

```diff
@@ -76,83 +76,34 @@ def processAlgorithm(self, progress):
-
-                outFeat.setAttributes(attrs)
-                outGeom = QgsGeometry(self.convertGeometry(multi_feature,
-                                                           vType))
-                outFeat.setGeometry(outGeom)
-                writer.addFeature(outFeat)
-
-            del writer
-        else:
-            raise GeoAlgorithmExecutionException(
-                self.tr('At least two features must have same attribute '
-                        'value! Please choose another field...'))
-
-    def singleToMultiGeom(self, wkbType):
-        try:
-            if wkbType in (QgsWkbTypes.Point, QgsWkbTypes.MultiPoint,
-                           QgsWkbTypes.Point25D, QgsWkbTypes.MultiPoint25D):
-                return QgsWkbTypes.MultiPoint
-            elif wkbType in (QgsWkbTypes.LineString, QgsWkbTypes.MultiLineString,
-                             QgsWkbTypes.MultiLineString25D,
-                             QgsWkbTypes.LineString25D):
-
-                return QgsWkbTypes.MultiLineString
-            elif wkbType in (QgsWkbTypes.Polygon, QgsWkbTypes.MultiPolygon,
-                             QgsWkbTypes.MultiPolygon25D, QgsWkbTypes.Polygon25D):
-
-                return QgsWkbTypes.MultiPolygon
-            else:
-                return QgsWkbTypes.Unknown
-        except Exception:
-            pass
-
-    def extractAsMulti(self, geom):
-        if geom.type() == QgsWkbTypes.PointGeometry:
-            if geom.isMultipart():
-                return geom.asMultiPoint()
-            else:
-                return [geom.asPoint()]
-        elif geom.type() == QgsWkbTypes.LineGeometry:
-            if geom.isMultipart():
-                return geom.asMultiPolyline()
-            else:
-                return [geom.asPolyline()]
-        else:
-            if geom.isMultipart():
-                return geom.asMultiPolygon()
-            else:
-                return [geom.asPolygon()]
-
-    def convertGeometry(self, geom_list, vType):
-        if vType == QgsWkbTypes.PointGeometry:
-            return QgsGeometry().fromMultiPoint(geom_list)
-        elif vType == QgsWkbTypes.LineGeometry:
-            return QgsGeometry().fromMultiPolyline(geom_list)
-        else:
-            return QgsGeometry().fromMultiPolygon(geom_list)
+        for inFeat in features:
+            atMap = inFeat.attributes()
+            idVar = atMap[index]
+            key = str(idVar).strip()
+            if not key in collection_geom.keys():
+                collection_geom[key] = []
+                collection_attrs[key] = atMap
+
+            inGeom = inFeat.geometry()
+            vType = inGeom.type()
+            collection_geom[key].append(inGeom)
+
+            current += 1
+            progress.setPercentage(int(current * total))
+
+        for key, geoms in collection_geom.items():
+            outFeat.setAttributes(collection_attrs[key])
+            outGeom = QgsGeometry().collectGeometry(geoms)
```

Comment:
You've still got the extra () in there. It's a static method, so isn't called on an object